### PR TITLE
Setup eslint for react

### PR DESCRIPTION
**Pull request by shinima**
I found that eslint fails to run in examples folder. So I install `eslint-plugin-react` and eslint now will lint the examples folder too. I'll update the code in example folder gradually.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "eslint:recommended",
+  "extends": ["eslint:recommended", "plugin:react/recommended"],
```
Comment (restrry) [inline .eslintrc.json]: should we add `"plugin:react/recommended"` only for examples folder?

Comment (shinima) [reply .eslintrc.json]: Agree. There are no react code in packages folder.

**Comment by restrry** [comment]
ok

**Comment by shinima** [comment]
Rebased this upon master.

**Pull request merged**